oibackoff/index.d.ts

```diff
@@ -0,0 +1,65 @@
+    <A,B>(fn: (callback: (a: A, b?: B) => any) => any, data0: any, data1: any, data2: any, data3: any, data4: any, data5: any, data6: any, intermediate: BackoffIntermediate<A>, callback: (a: A, b: (B | null), priorErrors?: A[]) => void): void;
+    <A,B>(fn: (callback: (a: A, b?: B) => any) => any, data0: any, data1: any, data2: any, data3: any, data4: any, data5: any, data6: any, callback: (a: A, b: (B | null), priorErrors?: A[]) => void): void;
+    <A,B>(fn: (data0: any, callback: (a: A, b?: B) => any) => any, data0: any, data1: any, data2: any, data3: any, data4: any, data5: any, data6: any, intermediate: BackoffIntermediate<A>, callback: (a: A, b: (B | null), priorErrors?: A[]) => void): void;
+    <A,B>(fn: (data0: any, callback: (a: A, b?: B) => any) => any, data0: any, data1: any, data2: any, data3: any, data4: any, data5: any, data6: any, callback: (a: A, b: (B | null), priorErrors?: A[]) => void): void;
+    <A,B>(fn: (callback: (a: A, b?: B) => any) => any, data0: any, data1: any, data2: any, data3: any, data4: any, data5: any, data6: any, data7: any, intermediate: BackoffIntermediate<A>, callback: (a: A, b: (B | null), priorErrors?: A[]) => void): void;
+    <A,B>(fn: (callback: (a: A, b?: B) => any) => any, data0: any, data1: any, data2: any, data3: any, data4: any, data5: any, data6: any, data7: any, callback: (a: A, b: (B | null), priorErrors?: A[]) => void): void;
+    <A,B>(fn: (data0: any, callback: (a: A, b?: B) => any) => any, data0: any, data1: any, data2: any, data3: any, data4: any, data5: any, data6: any, data7: any, intermediate: BackoffIntermediate<A>, callback: (a: A, b: (B | null), priorErrors?: A[]) => void): void;
+    <A,B>(fn: (data0: any, callback: (a: A, b?: B) => any) => any, data0: any, data1: any, data2: any, data3: any, data4: any, data5: any, data6: any, data7: any, callback: (a: A, b: (B | null), priorErrors?: A[]) => void): void;
+    <A,B>(fn: (callback: (a: A, b?: B) => any) => any, data0: any, data1: any, data2: any, data3: any, data4: any, data5: any, data6: any, data7: any, data8: any, intermediate: BackoffIntermediate<A>, callback: (a: A, b: (B | null), priorErrors?: A[]) => void): void;
+    <A,B>(fn: (callback: (a: A, b?: B) => any) => any, data0: any, data1: any, data2: any, data3: any, data4: any, data5: any, data6: any, data7: any, data8: any, callback: (a: A, b: (B | null), priorErrors?: A[]) => void): void;
+    <A,B>(fn: (data0: any, callback: (a: A, b?: B) => any) => any, data0: any, data1: any, data2: any, data3: any, data4: any, data5: any, data6: any, data7: any, data8: any, intermediate: BackoffIntermediate<A>, callback: (a: A, b: (B | null), priorErrors?: A[]) => void): void;
+    <A,B>(fn: (data0: any, callback: (a: A, b?: B) => any) => any, data0: any, data1: any, data2: any, data3: any, data4: any, data5: any, data6: any, data7: any, data8: any, callback: (a: A, b: (B | null), priorErrors?: A[]) => void): void;
```

Comment:
Here's the relevant code:

```js
var args = Array.prototype.slice.call(arguments);
var fn = args.shift();
var callback = args.pop();
var intermediate = function() {};
if ( _.isFunction( _.last( args ) ) ) {
    intermediate = args.pop();
}
...
args.push(myCallback);
...
fn.apply(null, args);
```
If my interpretation of all this is correct, `fn` should take as many data (non-callback) parameters as there are pieces of data passed in. (In effect this library is just partially applying the function for you before consuming a function that takes just a callback.) Here there are 8 data arguments, but the function has only 1 data parameter.

(Also, we probably only need overloads up to 4 data parameters, lest the type declaration become half as long as the library itself!)